### ISSUE TITLE
Workspace Chosen event no longer emitted

### ISSUE THREAD

**Issue by theherk** [MLFlexer/smart_workspace_switcher.wezterm#29]
Since e7f765c7cddb the `smart_workspace_switcher.workspace_chosen` event is no longer emitted. It looks like execution never enters [this else block](https://github.com/MLFlexer/smart_workspace_switcher.wezterm/blob/0338a3d82a063f4e2b1f43685f73b57eeda2c11c/plugin/init.lua#L100-L117). I can look at this further, but not immediately.

**Comment by MLFlexer** [comment]
The event was renamed to `smart_workspace_switcher.workspace_switcher.chosen` instead of `smart_workspace_switcher.workspace_chosen` That might be the cause of your issue? If not, then please let me know 😄

**Comment by theherk** [comment]
I caught that but typed the wrong one here. The issue is that `id` and `label` are different in each block. So in the `else` block it needs to match `id` rather than `label`. I'm not sure that is ideal though, since they should probably be consistent in either scenario.

If you checkout [this branch](https://github.com/theherk/smart_workspace_switcher.wezterm/tree/debug-id-label), you'll see what I mean.

**Comment by MLFlexer** [comment]
I see now, you are right, it should be changed to `id`instead of `label`

The reason it is not the same in both cases is that the first block of the if-statement should be activated when the user chooses a path, i.e. creating a new workspace. The path entries have the `id` be the full path and the `label` is what the user will see, i.e. the full path, but where the home directory is replaced by `~/`. 
The label is also what is used as name when spawning the new workspaces. This last part is important, as the label is used to emit the event to the correct mux window.

The else block of the if-statement should have an already existing workspace. These entries will have the `id` be the name of the workspace, and the `label` be some formatted string based on the workspace name. Now when the event is emitted, the event is emitted to the mux window within the workspace with the same name as the `id` as the `label` for entries that reach this else-block might not have the same `label` as their associated workspace name.

See [InputSelector](https://wezfurlong.org/wezterm/config/lua/keyassignment/InputSelector.html?h=label) for more info on labels and ids

TLDR: the label in the first block is the name of the created workspace, and the id in the else block is the name of an existing workspace.

I hope this explanation makes sense :smile:

**Comment by theherk** [comment]
It does, and I'm happy to change it. I just didn't want to do so as a bandaid if you had something else in mind. I'll open up the PR right now.

**Comment by MLFlexer** [comment]
Thank you for finding this, and even better fixing it! This is very much appreciated
